app/services/form_validator.rb

```diff
@@ -1,83 +1,92 @@
+  def validate_length(value, form_element)
+    if form_element[:data_type] == 'text' && (value || []).size >= MAX_LENGTH[:TEXT]
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_length', length: MAX_LENGTH[:TEXT])
+    elsif form_element[:data_type] == 'paragraph' && (value || []).size >= MAX_LENGTH[:PARAGRAPH]
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_length', length: MAX_LENGTH[:PARAGRAPH])
+    end
   end
 
-  def validate_required(form_element, el_name)
-    return nil unless form_element.required? && @params[el_name].blank?
-    @errors[el_name] << I18n.t('validation.is_required')
+  def validate_required(value, form_element)
+    return unless form_element[:required] && value.blank?
+    @errors[form_element[:name]] << I18n.t('validation.is_required')
   end
 
-  def validate_phone(form_element, el_name)
-    phone_number = @params[el_name]
-    if form_element.data_type == 'phone' && phone_number.present? && !is_phone(phone_number)
-      @errors[el_name] << I18n.t('validation.is_invalid_phone')
+  def validate_phone(phone_number, form_element)
+    if form_element[:data_type] == 'phone' && phone_number.present? && !is_phone(phone_number)
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_phone')
     end
   end
 
-  def validate_email(form_element, el_name)
-    email = @params[el_name].try(:encode, 'UTF-8', invalid: :replace, undef: :replace)
-    if form_element.data_type == 'email' && email.present? && !is_email(email)
-      @errors[el_name] << I18n.t('validation.is_invalid_email')
+  def validate_email(value, form_element)
+    email = value.try(:encode, 'UTF-8', invalid: :replace, undef: :replace)
+    if form_element[:data_type] == 'email' && email.present? && !is_email(email)
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_email')
     end
   end
 
-  def validate_country(form_element, el_name)
-    country = @params[el_name]
-    if form_element.data_type == 'country' && country.present? && !is_country_code(country)
-      @errors[el_name] << I18n.t('validation.is_invalid_country')
+  def validate_country(country, form_element)
+    if form_element[:data_type] == 'country' && country.present? && !is_country_code(country)
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_country')
     end
   end
 
-  def validate_postal(form_element, el_name)
-    postal = @params[el_name]
+  def validate_postal(postal, form_element)
     country = (@params[:country].blank? ? :US : @params[:country].to_sym)
 
-    if form_element.data_type == 'postal' && postal.present? && !is_postal(postal, country)
-      @errors[el_name] << I18n.t('validation.is_invalid_postal')
+    if form_element[:data_type] == 'postal' && postal.present? && !is_postal(postal, country)
+      @errors[form_element[:name]] << I18n.t('validation.is_invalid_postal')
     end
   end
 
```

Comment:
I'm not sure if this FormValidator class is used in any other way, but from what I can see in the controller the only method that gets called on a FormValidator object is `valid?`. If this is the case, I'd make every other method `private` to make the class API more explicit.